### PR TITLE
fix: Prevent NullReferenceExeption on APIGatewayProxyRequest

### LINKED ISSUE #2528: Error occurs with AWS Lambda when sending non-standard APIGatewayProxyRequest json.
**Description**
Lambda function: APIGatewayProxyRequest 

When sending the following json to an APIGatewayProxyRequest  lambda function, it fails with System.ArgumentNullException errors.

```
{
  "httpMethod": "GET",
  "path": "/test",
  "body": "{\"Name\":\"John\"}"
}
```

Exceptions:

```
2024-06-06 15:21:33,716 NewRelic   WARN: [pid: 16, tid: 9] Unexpected exception in AddEventTypeAttributes(). Event type APIGatewayProxyRequest had an inputObject of type Amazon.Lambda.APIGatewayEvents.APIGatewayProxyRequest. Exception: System.ArgumentNullException: Value cannot be null. (Parameter 'parameters')
   at NewRelic.Agent.Core.Transactions.Transaction.SetRequestParameters(IEnumerable`1 parameters)
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
   at NewRelic.Agent.Extensions.Lambda.LambdaEventHelpers.SetWebRequestProperties(IAgent agent, ITransaction transaction, Object webReqEvent)
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid4[T0,T1,T2,T3](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
   at NewRelic.Agent.Extensions.Lambda.LambdaEventHelpers.AddEventTypeAttributes(IAgent agent, ITransaction transaction, AwsLambdaEventType eventType, Object inputObject)
```

```
2024-06-06 15:21:33,734 NewRelic  ERROR: [pid: 16, tid: 9] Tracer invocation error
System.ArgumentNullException: Value cannot be null. (Parameter 'parameters')
   at NewRelic.Agent.Core.Transactions.Transaction.SetRequestParameters(IEnumerable`1 parameters)
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
   at NewRelic.Agent.Extensions.Lambda.LambdaEventHelpers.SetWebRequestProperties(IAgent agent, ITransaction transaction, Object webReqEvent)
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid4[T0,T1,T2,T3](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
   at NewRelic.Agent.Extensions.Lambda.LambdaEventHelpers.AddEventTypeAttributes(IAgent agent, ITransaction transaction, AwsLambdaEventType eventType, Object inputObject)
   at NewRelic.Providers.Wrapper.AwsLambda.HandlerMethodWrapper.BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall, IAgent agent, ITransaction transaction)
   at NewRelic.Agent.Core.Wrapper.WrapperService.BeforeWrappedMethod(Type type, String methodName, String argumentSignature, Object invocationTarget, Object[] methodArguments, String tracerFactoryName, String metricName, UInt32 tracerArguments, UInt64 functionId)
   at NewRelic.Agent.Core.AgentManager.GetTracerImpl(String tracerFactoryName, UInt32 tracerArguments, String metricName, String assemblyName, Type type, String typeName, String methodName, String argumentSignature, Object invocationTarget, Object[] arguments, UInt64 functionId)
```

The problem appears to be in [this method](https://github.com/newrelic/newrelic-dotnet-agent/blob/2bfc317c5154461a2b9cdd8abd6134dc32afbbfe/src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Lambda/LambdaEventHelpers.cs#L295) on line 295.

**Expected Behavior**
It just works, or at least an null exceptions are handled.

**Steps to Reproduce**
Here is the meat of the lambda function:

```
public APIGatewayProxyResponse FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context) => new APIGatewayProxyResponse()
{
    StatusCode = 200,
    Body = Environment.GetEnvironmentVariable(new Guid().ToString())
};
```

Publish this, add the layer, and send the initial JSON and the error occurs.

**For Maintainers Only or Hero Triaging this bug**
P2
S

### PR DESCRIPTION
**Pull request by tippmar-nr**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

This fixes a null reference exception that could occur when a minimal JSON payload was sent to to a Lambda function that take an `APIGatewayProxyRequest` or `APIGatewayHttpApiV2ProxyRequest`.

Includes integration tests to validate the fix.

Fixes #2528

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: tests/Agent/IntegrationTests/IntegrationTests/AwsLambda/AwsLambdaAPIGatewayHttpApiV2ProxyRequestTest.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ protected AwsLambdaAPIGatewayHttpApiV2ProxyRequestTest(T fixture, ITestOutputHel
                 {
                     _fixture.EnqueueAPIGatewayHttpApiV2ProxyRequest();
                     _fixture.EnqueueAPIGatewayHttpApiV2ProxyRequestWithDTHeaders(TestTraceId, TestParentSpanId);
+                    _fixture.EnqueueMinimalAPIGatewayHttpApiV2ProxyRequest();
                     _fixture.AgentLog.WaitForLogLines(AgentLogBase.ServerlessPayloadLogLineRegex, TimeSpan.FromMinutes(1), 2);
```
Comment (nrcventura) [inline tests/Agent/IntegrationTests/IntegrationTests/AwsLambda/AwsLambdaAPIGatewayHttpApiV2ProxyRequestTest.cs:38]: I think that this line now needs to wait for 3 payloads to be generated.

File: tests/Agent/IntegrationTests/IntegrationTests/AwsLambda/AwsLambdaAPIGatewayRequestTest.cs [hunk 2]
```diff
@@ -34,6 +34,7 @@ protected AwsLambdaAPIGatewayProxyRequestTest(T fixture, ITestOutputHelper outpu
                 {
                     _fixture.EnqueueAPIGatewayProxyRequest();
                     _fixture.EnqueueAPIGatewayProxyRequestWithDTHeaders(TestTraceId, TestParentSpanId);
+                    _fixture.EnqueueMinimalAPIGatewayProxyRequest();
                     _fixture.AgentLog.WaitForLogLines(AgentLogBase.ServerlessPayloadLogLineRegex, TimeSpan.FromMinutes(1), 2);
```
Comment (nrcventura) [inline tests/Agent/IntegrationTests/IntegrationTests/AwsLambda/AwsLambdaAPIGatewayRequestTest.cs:38]: This line should also wait for 3 payloads now.

File: tests/Agent/IntegrationTests/IntegrationTests/AwsLambda/AwsLambdaAPIGatewayHttpApiV2ProxyRequestTest.cs (hunk 1 above)
Comment (tippmar-nr) [reply tests/Agent/IntegrationTests/IntegrationTests/AwsLambda/AwsLambdaAPIGatewayHttpApiV2ProxyRequestTest.cs:38]: Good catch. Tests were passing locally even without it but I'll make the update.

**Review by nr-ahemsath — APPROVED** [review]
Looks good!

**Review by nrcventura — APPROVED** [review]

**Pull request merged**